lib/rules/selector-no-qualifying-type/index.mjs

```diff
@@ -99,45 +247,54 @@ const rule = (primary, secondaryOptions) => {
 			}
 
 			/**
-			 * @param {Container<string | undefined>} resolvedSelectorNode
+			 * @param {Root} resolvedSelectorNode
 			 * @param {Selector} selectorNode
 			 * @param {boolean} nested
 			 */
 			function checkSelector(resolvedSelectorNode, selectorNode, nested) {
-				resolvedSelectorNode.walkTags((tagNode) => {
-					if (!isStandardSyntaxTypeSelector(tagNode)) return;
+				const { compoundSelectors, lookupMap } = getCompoundSelectors(resolvedSelectorNode);
+
+				compoundSelectors.walkTags((tagNode) => {
+					const originalTag = lookupMap.get(tagNode);
+
+					if (!originalTag) return;
+
+					if (!isStandardSyntaxTypeSelector(originalTag)) return;
 
 					const selectorParent = tagNode.parent;
 
-					if (selectorParent && selectorParent.nodes.length === 1) {
+					if (!selectorParent || selectorParent.nodes.length === 1) {
 						return;
 					}
 
-					const rightNodes = getRightNodes(tagNode);
+					const siblingNodes = selectorParent.nodes;
 
-					for (const rightNode of rightNodes) {
+					const originalCompound = getOriginalCompound(originalTag);
+
+					for (const siblingNode of siblingNodes) {
 						if (
-							(rightNode.type === 'id' && !ignoreId) ||
-							(rightNode.type === 'class' && !ignoreClass) ||
-							(rightNode.type === 'attribute' && !ignoreAttribute)
+							(siblingNode.type === 'id' && !ignoreId) ||
+							(siblingNode.type === 'class' && !ignoreClass) ||
+							(siblingNode.type === 'attribute' && !ignoreAttribute)
 						) {
+							const selector = originalCompound.join('').trim();
+
 							let index = 0;
 							let endIndex = 0;
-							const selector = [tagNode, ...rightNodes].join('').trim();
 
 							if (nested) {
 								({ index, endIndex } = getStrippedSelectorSource(selectorNode));
 							} else {
 								index = tagNode.sourceIndex ?? 0;
-								endIndex = index + selector.length;
+								endIndex = index + tagNode.value.length;
 							}
 
 							report({
 								ruleName,
 								result,
 								node: ruleNode,
 								message: messages.rejected,
-								messageArgs: [selector],
+								messageArgs: [tagNode.toString().trim(), selector],
```

Comment:
[suggestion] Seeing L289 (`endIndex = index + tagNode.value.length`), using `tagNode.value` seems to be more consistent. Reasonable?


```suggestion
								messageArgs: [tagNode.value, selector],
```